main.py

```diff
@@ -136,19 +199,29 @@ def main():
         path_orientation_flag = move_player(player_pos, keys, paths, player_speed)
         if path_orientation_flag == "horizontal" or path_orientation_flag == "vertical":
             path_orientation = path_orientation_flag
+
         screen.fill(BLACK)
         draw_paths(screen, BLUE, paths)
         draw_player(screen, player_color, player_pos, player_size, path_orientation)
         draw_nodes(screen, node_color, nodes, node_size)
 
         for node_pos in nodes[:]:
             if detect_collision(player_pos, node_pos, player_size, node_size):
-                if ask_question_node():
-                    nodes.remove(node_pos)
-                    print(f"Problem solved at node: {node_problems[tuple(node_pos)]}")
-                else:
-                    print("You Need To Come Back")
-
+                if node_pos in node_problems:
```

Comment:
I think this is good logic you implemented for checking if a a node has been crossed by a user, however, this implementation becomes very difficult for other people to add different white hat challenges on to the game. So perhaps create a class that has these validations for this specific challenge then give that class a run() and validate method such that it contains all the logic to check the user has entered the correct information or not.

Reply:
okay ill try to make changes